src/directives/for-of/for_of.directive.ts

```diff
@@ -90,15 +89,15 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
             totalWidth = this.initHCache(this.igxForOf);
             this.hScroll = this.getElement(vc, "igx-horizontal-virtual-helper");
             if (this.hScroll) {
-                this._currIndex = this.getHorizontalIndexAt(this.hScroll.scrollLeft, this.hCache, 0);
+                this.state.startIndex = this.getHorizontalIndexAt(this.hScroll.scrollLeft, this.hCache, 0);
             }
         }
-        this._pageSize = this._calculatePageSize();
+        this.state.chunkSize = this._calculatePageSize();
```

Comment:
@dkamburov How about renaming: "_calculatePageSize" -> "_calculateChunkSize" and maybe every other method that contains "page" in its name.